api/api.ts

```diff
@@ -0,0 +1,244 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+
+/**
+ * API version information.
+ */
+export enum Version {
+    v0 = 0,         // 0.x.x
+    latest = v0,
+}
+
+/**
+ * The interface provided by the CMake Tools extension during activation.
+ * It is recommended to use the helper function [getCMakeToolsApi](#getCMakeToolsApi)
+ * instead of querying the extension instance directly.
+ */
+export interface CMakeToolsExtensionExports {
+    /**
+     * Get an API object.
+     * @param version The desired API version.
+     */
+    getApi(version: Version): CMakeToolsApi;
+}
+
+/**
+ * An interface to allow VS Code extensions to interact with the CMake Tools extension.
+ */
+export interface CMakeToolsApi {
+    /**
+     * The version of the API.
+     */
+    readonly version: Version;
+
+    /**
+     * Shows the given UI element.
+     * @param element Element to show.
+     */
+    showUiElement(element: UiElement): Promise<void>;
+
+    /**
+     * Hides the given UI element.
+     * @param element Element to hide.
+     */
+    hideUiElement(element: UiElement): Promise<void>;
+
+    /**
+     * An event that fires when the selected build target changes.
+     */
+    readonly onBuildTargetNameChanged: vscode.Event<string>;
+
+    /**
+     * An event that fires when the selected launch target changes.
+     */
+    readonly onLaunchTargetNameChanged: vscode.Event<string>;
+
+    /**
+     * An event that fires when the active folder changes.
+     */
+    readonly onActiveFolderChanged: vscode.Event<vscode.WorkspaceFolder | undefined>;
```

Comment:
I've renamed the API calls to refer to projects and use URIs, but I didn't see any code that tracks the active project. Currently the implmentation internally is still using the URIs of workspace folders. Is that correct for now, or did I miss something?

Reply:
You are correct.  Support for multiple projects in a single workspace folder is currently under development.  It will we available soon.  I just didn't want us to add a method that would immediately be deprecated.